test/cppsuite/tests/search_near_02.cxx

```diff
@@ -111,7 +111,7 @@ class search_near_02 : public test_harness::test {
                 if (tc->insert(cc.cursor, cc.coll.id, key)) {
                     if (tc->transaction.can_commit())
                         /* We are not checking the result of commit as it is not necessary. */
-                        tc->transaction.commit();
+                        WT_IGNORE_RET(tc->transaction.commit());
```

Comment:
Here I am fine ignoring the result as I am inserting random stuff. And if this one fails a couple of times, it won't impact much the outcome.
However, if it never commits, that would be an issue...
I remember we suggested having a maximum number of failing calls.

Reply:
Is it possible that can_commit() returns true, but commit() returns false? Should that edge case even be handled?

Reply:
It is possible given our current implementation. I think at this point we should probably check that `transaction_context::commit` returns either `0` or `WT_ROLLBACK`, I think this is what is returned if the commit needs to be rolled back internally? See [here](https://github.com/wiredtiger/wiredtiger/blob/develop/test/cppsuite/test_harness/workload/thread_context.cxx#L85-L88) and [there](https://github.com/wiredtiger/wiredtiger/blob/develop/src/session/session_api.c#L1641-L1658).